### PR TITLE
Add smoke test for NativeAot.

### PR DESCRIPTION
**Pull request by tmds**
@omajid ptal.

### CODE REVIEW

File: nativeaot/test.sh [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+fi
+
+# Verify the published output contains a single file.
+PUBLISHED_FILE_COUNT=$(ls ./bin/Debug/net*/*/publish | wc -l)
+if [ "$PUBLISHED_FILE_COUNT" != "1" ]; then
+    echo "FAIL: published NativeAot app contains more than 1 file."
+    exit 1
+fi
+
+# Verify the application runs successfully.
```
Comment (omajid) [inline nativeaot/test.sh:22]: Would it be useful to verify the application also produces the expected output and no other warnings/output?

**Comment by omajid** [comment]
THANKS!!!!

File: nativeaot/test.sh (hunk 1 above)
Comment (tmds) [reply nativeaot/test.sh]: I've updated the test to verify the application output instead of checking the exit code.

**Comment by tmds** [comment]
Is this good to merge @omajid ?

**Comment by omajid** [comment]
Merged now!

**Pull request merged**